test.ps1

```diff
@@ -25,89 +25,100 @@ param (
 # Import shared functionality
 . ./shared.ps1
 
-$buildDir = Join-Path $PSScriptRoot "build"
-$practiceExercisesDir = Join-Path $buildDir "practice"
-$conceptExercisesDir = Join-Path $buildDir "concept"
-$sourceDir = Resolve-Path "exercises"
 
-function Configlet-Lint {
+function Invoke-Configlet-Lint {
     Write-Output "Linting config.json"
-    Run-Command "./bin/fetch-configlet"
-    Run-Command "./bin/configlet lint"
+    Invoke-CommandExecution "./bin/fetch-configlet"
+    Invoke-CommandExecution "./bin/configlet lint"
 }
 
-function Build-Generators { 
+function Invoke-DotNetBuild-Generators { 
     Write-Output "Building generators"
-    Run-Command "dotnet build ./generators"
+    Invoke-CommandExecution "dotnet build ./generators"
 }
 
-function Clean {
+function Clean($BuildDir) {
     Write-Output "Cleaning previous build"
-    Remove-Item -Recurse -Force $buildDir -ErrorAction Ignore
+    Remove-Item -Recurse -Force $BuildDir -ErrorAction Ignore
 }
-
-function Copy-Exercises {
+function Copy-Exercise($SourceDir, $BuildDir) {
     Write-Output "Copying exercises"
-    Copy-Item $sourceDir -Destination $buildDir -Recurse
+    Copy-Item $SourceDir -Destination $BuildDir -Recurse
 }
 
-function Enable-All-Tests {
+function ConvertTo-UnSkippedTest($BuildDir) {
     Write-Output "Enabling all tests"
-    Get-ChildItem -Path $buildDir -Include "*Tests.cs" -Recurse | ForEach-Object {
+    Get-ChildItem -Path $BuildDir -Include "*Tests.cs" -Recurse | ForEach-Object {
         (Get-Content $_.FullName) -replace "Skip = ""Remove this Skip property to run this test""", "" | Set-Content $_.FullName
     }
 }
 
-function Test-Refactoring-Projects {
-    Write-Output "Testing refactoring projects"
-    @("tree-building", "ledger", "markdown") | ForEach-Object { Run-Command "dotnet test $practiceExercisesDir/$_" }
+function Test-ExeciseWithRefactoring($PracticeExercisesDir) {
```

Comment:
Why this naming change?

Reply:
Test-Refactoring-Projects. Project is not singular.
new names could be:
Test-Refactoring-Exercise
Test-Refactoring-Project

Reply:
I can't come up with a better name, so I'd prefer to ignore the naming rule entirely. I think it leads to less easy to understand code, so I think we should just completely ignore the naming rule and revert the naming changes.

I see that the analyzer supports a settings file to ignore rules: https://github.com/PowerShell/PSScriptAnalyzer#explicit